### PR TITLE
WebApp add more CSP and security headers

### PR DESCRIPTION
**Pull request by raix**
### Summary & Motivation

Add more Content Security Policies and security headers to improve security and lighthouse scores

### Checklist

- [x] I have added a Label to the pull-request
- [x] I have added tests, and done manual regression tests
- [x] I have updated the documentation, if necessary

### CODE REVIEW

File: application/shared-kernel/ApiCore/Middleware/WebAppMiddleware.cs [hunk 1]
```diff
@@ -119,6 +127,23 @@ private string GetHtmlWithEnvironment(Dictionary<string, string>? requestEnviron
...
+    }
+
+    private static void ApplySecurityHeaders(HttpContext context, StringValues contentSecurityPolicy)
+    {
+        context.Response.Headers.Append("Content-Security-Policy", contentSecurityPolicy);
+        // Disable automatic content type detection
+        context.Response.Headers.Append("X-Content-Type-Options", "nosniff");
+        context.Response.Headers.Append("X-Frame-Options", "DENY");
+        context.Response.Headers.Append("X-XSS-Protection", "1; mode=block");
+        context.Response.Headers.Append("Strict-Transport-Security", "max-age=31536000; includeSubDomains");
```
Comment (tjementum) [inline application/shared-kernel/ApiCore/Middleware/WebAppMiddleware.cs:144]: The HTTPS strict header is handled by Program.cs. I believe most if not alle of these should apply to both API and Index.html end-points.

Comment (raix) [reply application/shared-kernel/ApiCore/Middleware/WebAppMiddleware.cs:144]: Ok - I can remove the ApplySecurityHeaders then

Comment (tjementum) [reply application/shared-kernel/ApiCore/Middleware/WebAppMiddleware.cs:144]: I'm a bit in doubt. Maybe it's only "trict-Transport-Security". Not sure if it makes sense to see X-Frame-Options and X-XSS-Protection on API calls. Maybe these do belong here as they are only for HTML?

Comment (raix) [reply application/shared-kernel/ApiCore/Middleware/WebAppMiddleware.cs]: I've removed it for now - I'm sure we will revisit this (I constantly look at lighthouse and security issues / audit in the browser)

**Pull request merged**